src/server/input.ts

```diff
@@ -50,14 +51,98 @@ export class Keyboard {
     this._page = page;
   }
 
+  async _updateKeyboardIndicator(description?: KeyDescription) {
```

Comment:
Let's get some code modularity going. This is a debugging code, it needs to be strictly separate from the code functionality. See what Dmitry did for tracing - make sure your auxiliary feature is using apis / dependency injection where necessary.